docs/operating-scylla/procedures/cluster-management/handling-membership-change-failures.rst

```diff
@@ -0,0 +1,202 @@
+
+---------------------------
+Handling a failed bootstrap
+---------------------------
+
+If a failure happens when trying to bootstrap a new node to the cluster, you can try bootstrapping the node again by restarting it.
+
+If the node refuses to bootstrap on subsequent attempts, follow the instructions in the :ref:`cleaning up after a failed membership change <cleaning-up-after-change>` section, then clear the node's data directories before trying to bootstrap it again.
+
+------------------------------
+Handling a failed decommission
+------------------------------
+
+There are two cases.
+
+Most likely the failure happened during the data repair/streaming phase - before the node tried to leave the token ring. Look for a log message containing "leaving token ring" in the logs of the node that you tried to decommission. For example:
+
+.. code-block:: console
+
+    INFO  2023-03-14 13:08:38,323 [shard 0] storage_service - decommission[5b2e752e-964d-4f36-871f-254491f4e8cc]: leaving token ring
+
+If the message is **not** present, the failure happened before the node tried to leave the token ring. In that case you can simply restart the node and attempt to decommission it again.
+
+If the message is present, the node attempted to leave the token ring, but it might have left the cluster only partially before the failure. **Do not try to restart the node**. Instead, you must remove any leftovers using the :doc:`removenode operation </operating-scylla/nodetool-commands/removenode/>`. See :ref:`cleaning up after a failed membership change <cleaning-up-after-change>`.
+
+If you don't have access to the node's logs anymore, assume the second case (the node might have attempted to leave the token ring), **do not try to restart the node**, instead follow the :ref:`cleaning up after a failed membership change <cleaning-up-after-change>` section.
+
+----------------------------
+Handling a failed removenode
+----------------------------
+
+Simply retry the removenode operation.
+
+If you somehow lost the host ID of the node that you tried to remove, follow the instructions in :ref:`cleaning up after a failed membership change <cleaning-up-after-change>`.
+
+--------------------------
+Handling a failed replace
+--------------------------
+
+Replace is a special case of bootstrap, but the bootstrapping node tries to take the place of another dead node. You can retry a failed replace operation by restarting the replacing node.
+
+If the node refuses to perform replace on subsequent attempts, follow the instructions in  :ref:`cleaning up after a failed membership change <cleaning-up-after-change>` section to remove the remains of the replacing node. You can then clear the node's data directories and attempt to replace again, or you can remove the dead node which you initially tried to replace using :doc:`removenode </operating-scylla/nodetool-commands/removenode/>`, then perform a regular bootstrap.
```

Comment:
I think "refuses" is hardly possible. In fact, if you know any reason for a restart to fail, other than a resource problem (disk, memory, OS resources such as file descriptors), it is best to mention such reason explicitly. Otherwise I think it is better to say "If the failure persists", or something along these lines.